accepted/0000-isolated-workspaces.md

```diff
@@ -0,0 +1,177 @@
+- Start Date: 2018-02-12
+- RFC PR: (leave this empty)
+- Yarn Issue: (leave this empty)
+
+# Summary
+
+Working on a workspace in isolation should be as easy as working on all workspaces together. Running a yarn command in
+a workspace should behave the same whether it is part of a larger project or whether it is its own separate repo. There
+should be a new set of commands, `yarn workspaces <command>` to distinguish between running commands that affect
+all workspaces from commands that affect a single workspace.
+
+# Motivation
+
+Currently, yarn workspaces are optimized for people making changes across multiple projects in the 
+monorepo, but they can make things more difficult if you want to work on a single workspace in isolation. While
+the automatic linking and auto-installation of dependencies for all workspaces can be convenient if you are actually working
+on all workspaces, having to build and install everything locally for multiple projects when you only work on one
+at a time can be time consuming compared to just pulling those packages down from a registry.
+
+If you could work on a single workspace in isolation as easily as if it were its own repo, the transition to a monorepo
+from separate repos would be seamless for people who focus on a single workspace/repo at a time, and one of the
+biggest reasons not to use workspaces would be removed. 
+
+# Detailed design
+
+Terminology used for this doc:
+- rootDir: the root directory of a workspaces project. the one with the package.json with workspaces in it
+- workspaceDir: Directory for an individual workspace/package
+- isolated mode: `yarn foo`
+- workspaces mode `yarn workspaces foo` (could also be `yarn foo --workspaces` or `yarn foo -W` like yarn add)
```

Comment:
So this is the default mode right now and you propose isolated mode to become default?

I would rather avoid a breaking change and reverse it.
Via .yarnrc you can set up a subfolder to work in isolated mode and if a user cd to that folder Yarn would work in isolated mode.
But if user cd to any other folder Yarn would work the way it works right now

Reply:
That works. I did not know about settings flags via yarnrc. I do think it's worth reconsidering the default mode at the next major version, but that can be a separate issue.

So if i used that, i could still go into the workspace and say `yarn install --isolate=false` to run explicitly in workspaces mode? That is a bit awkward. Kinda why i was thinking of having a --workspaces flag that is basically the opposite of --isolate.

Reply:
I propose doing a non breaking progressive change first and shipping it.
Then before Yarn v2.0 we can ask people to weigh in if introducing the reverse would be better for them.

Reply:
fair enough! I'll make those changes to the doc